File: sdks/akta-docs/csharp/Diagnostic.cs
using System.Text.Json.Serialization;

namespace OpenAkta.AktaDocs;

public sealed record Diagnostic(
    [property: JsonPropertyName("file")] string File,
    [property: JsonPropertyName("line")] int Line,
    [property: JsonPropertyName("column")] int Column,
    [property: JsonPropertyName("rule_id")] string RuleId,
    [property: JsonPropertyName("severity")] string Severity,
    [property: JsonPropertyName("message")] string Message,
    [property: JsonPropertyName("end_line")] int? EndLine,
    [property: JsonPropertyName("end_column")] int? EndColumn);

File: sdks/akta-docs/java/src/main/java/dev/openakta/aktadocs/Diagnostic.java
package dev.openakta.aktadocs;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record Diagnostic(
        String file,
        int line,
        int column,
        @JsonProperty("rule_id") String ruleId,
        String severity,
        String message,
        @JsonProperty("end_line") Integer endLine,
        @JsonProperty("end_column") Integer endColumn,
        @JsonProperty("doc_url") String docUrl
) {
    public Diagnostic(
            String file,
            int line,
            int column,
            String ruleId,
            String severity,
            String message,
            Integer endLine,
            Integer endColumn
    ) {
        this(file, line, column, ruleId, severity, message, endLine, endColumn, null);
    }
}
